test/testdrive/subscribe-output.td

```diff
@@ -268,3 +268,121 @@ DELETE FROM t2 WHERE key = 1
 
 ! SUBSCRIBE (SELECT * FROM t2) WITH (PROGRESS, SNAPSHOT) ENVELOPE UPSERT (KEY (invalid_key))
 contains: column "invalid_key" does not exist
+
+
+> BEGIN
+> DECLARE c CURSOR FOR SUBSCRIBE (SELECT 1 as a, 2 as b union select 2, 1) ENVELOPE DEBEZIUM (KEY (a, b))
+> FETCH 1 c
+<TIMESTAMP> insert 1 2
+
+> FETCH 1 c
+<TIMESTAMP> insert 2 1
+> COMMIT
+
+> BEGIN
+> DECLARE c CURSOR FOR SUBSCRIBE (SELECT 1 as a, 2 as b union select 2, 1) ENVELOPE DEBEZIUM (KEY (b, a))
+> FETCH 1 c
+<TIMESTAMP> insert 1 2
+
+> FETCH 1 c
+<TIMESTAMP> insert 2 1
+> COMMIT
+
+> DROP TABLE t1 CASCADE
+> CREATE TABLE t1(key int, value int, UNIQUE (key))
+> INSERT INTO t1 VALUES (1, 2), (2, 3)
+
+> BEGIN
+> DECLARE c CURSOR FOR SUBSCRIBE (SELECT * FROM t1) ENVELOPE DEBEZIUM (KEY (key))
+> FETCH 2 c
+<TIMESTAMP> insert 1 <null> 2
+<TIMESTAMP> insert 2 <null> 3
+
+$ postgres-execute connection=mz_system
+UPDATE t1 SET value = 4 WHERE key = 1
+
+> FETCH 1 c
+<TIMESTAMP> upsert 1 2 4
+
+$ postgres-execute connection=mz_system
+DELETE FROM t1 WHERE key = 1
+
+> FETCH ALL c
+<TIMESTAMP> delete 1 4 <null>
+
+$ postgres-execute connection=mz_system
+INSERT INTO t1 VALUES (1, 1), (1, 1)
+
+> FETCH ALL c
+<TIMESTAMP> "key_violation" 1 <null> <null>
+
+$ postgres-execute connection=mz_system
+DELETE FROM t1 WHERE key = 1
+
+> FETCH ALL c
+<TIMESTAMP> "key_violation" 1 <null> <null>
+
+$ postgres-execute connection=mz_system
+INSERT INTO t1 VALUES (1, 1), (1, 2)
+
+> FETCH ALL c
+<TIMESTAMP> "key_violation" 1 <null> <null>
+
+$ postgres-execute connection=mz_system
+INSERT INTO t1 VALUES (1, 3)
+
+> FETCH ALL c
+<TIMESTAMP> insert 1 <null> 3
+
+$ postgres-execute connection=mz_system
+DELETE FROM t1 WHERE key = 1 AND value = 1
+
+> FETCH ALL c
+<TIMESTAMP> delete 1 1 <null>
+
+$ postgres-execute connection=mz_system
+DELETE FROM t1 WHERE key = 1
+
+> FETCH ALL c
+<TIMESTAMP> "key_violation" 1 <null> <null>
+
+> COMMIT
+
+> DROP TABLE t2 CASCADE
+> CREATE TABLE t2(value int, key int, UNIQUE (key))
+> INSERT INTO t2 VALUES (2, 1), (3, 2)
+
+> BEGIN
+> DECLARE c CURSOR FOR SUBSCRIBE (SELECT * FROM t2) ENVELOPE DEBEZIUM (KEY (key))
+> FETCH 2 c
+<TIMESTAMP> insert 1 <null> 2
+<TIMESTAMP> insert 2 <null> 3
+
+$ postgres-execute connection=mz_system
+UPDATE t2 SET value = 4 WHERE key = 1
+
+> FETCH 1 c
+<TIMESTAMP> upsert 1 2 4
+
+$ postgres-execute connection=mz_system
+DELETE FROM t2 WHERE key = 1
+
+> FETCH ALL c
+<TIMESTAMP> delete 1 4 <null>
+
+> COMMIT
+
+# make sure progress messages are ok
+> BEGIN
+> DECLARE c CURSOR FOR SUBSCRIBE (SELECT * FROM t2) WITH (PROGRESS, SNAPSHOT) ENVELOPE DEBEZIUM (KEY (key))
+> FETCH 1 c
+<TIMESTAMP> true <null> <null> <null> <null>
+
+> FETCH ALL c
```

Comment:
This unfortunately is timing sensitive, sometimes `FETCH ALL` will only return 1 row, if the second timestamp is not yet available , please use `FETCH 2` instead, see https://github.com/MaterializeInc/materialize/pull/19107

Reply:
ah thanks, fixed everywhere